### PR TITLE
[fix] Sonarqube - Added more changes to make PR code coverage work

### PR DESCRIPTION
**Pull request by neteinstein**

### CODE REVIEW

File: base/src/main/java/com/mindera/skeletoid/threads/threadpools/NamedThreadFactory.kt [hunk 1]
```diff
@@ -20,6 +20,24 @@ open class NamedThreadFactory internal constructor(
...
+        private const val LOG_TAG = "NamedThreadFactory"
+    }
+
+    /**
+     * Default constructor
+     *
+     * @param threadPoolName The name of the ThreadPool
+     */
+    init {
+        val s = System.getSecurityManager()
```
Comment (bruno-garces) [inline base/src/main/java/com/mindera/skeletoid/threads/threadpools/NamedThreadFactory.kt]: Can we have a meaningful name?

File: base/src/test/java/com/mindera/skeletoid/threads/threadpools/NamedThreadFactoryUnitTest.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    @Test
+    fun testGetThreads() {
+        Assert.assertNotNull(namedThreadFactory.threads)
+        Assert.assertEquals(0, namedThreadFactory.threads.size)
+    }
+
+    @Test
+    fun testNewThreads() {
+        val runnable =
+            Mockito.mock(Runnable::class.java)
```
Comment (bruno-garces) [inline base/src/test/java/com/mindera/skeletoid/threads/threadpools/NamedThreadFactoryUnitTest.kt]:
```suggestion
        val runnable = Mockito.mock(Runnable::class.java)
```

File: base/src/test/java/com/mindera/skeletoid/threads/threadpools/ScheduledThreadPoolUnitTest.kt [hunk 3]
```diff
@@ -15,15 +19,15 @@ class ScheduledThreadPoolUnitTest {
 
     @Test
     fun testShutdownThreadPool() {
-        val namedThreadFactory : NamedThreadFactory = mock()
+        val namedThreadFactory : NamedThreadFactory = NamedThreadFactory("",1)
```
Comment (bruno-garces) [inline base/src/test/java/com/mindera/skeletoid/threads/threadpools/ScheduledThreadPoolUnitTest.kt]:
```suggestion
        val namedThreadFactory : NamedThreadFactory = NamedThreadFactory("", 1)
```

File: base/src/test/java/com/mindera/skeletoid/threads/threadpools/ScheduledThreadPoolUnitTest.kt [hunk 4]
```diff
@@ -15,15 +19,15 @@ class ScheduledThreadPoolUnitTest {
...
         val scheduledThreadPoolExecutor = ScheduledThreadPoolExecutor(1, namedThreadFactory)
         scheduledThreadPoolExecutor.shutdown()
         Assert.assertTrue(scheduledThreadPoolExecutor.isShutdown)
     }
 
     @Test
     fun testShutdownNowThreadPool() {
-        val namedThreadFactory : NamedThreadFactory = mock()
+        val namedThreadFactory : NamedThreadFactory = NamedThreadFactory("",1)
```
Comment (bruno-garces) [inline base/src/test/java/com/mindera/skeletoid/threads/threadpools/ScheduledThreadPoolUnitTest.kt]:
```suggestion
        val namedThreadFactory : NamedThreadFactory = NamedThreadFactory("", 1)
```

File: base/src/test/java/com/mindera/skeletoid/threads/threadpools/ScheduledThreadPoolUnitTest.kt [hunk 5]
```diff
@@ -3,8 +3,12 @@ package com.mindera.skeletoid.threads.threadpools
 import com.mindera.skeletoid.utils.extensions.mock
 import org.junit.Assert
 import org.junit.Test
+import org.mockito.Mockito
+import java.util.concurrent.ArrayBlockingQueue
 
 class ScheduledThreadPoolUnitTest {
+
+
```
Comment (bruno-garces) [inline base/src/test/java/com/mindera/skeletoid/threads/threadpools/ScheduledThreadPoolUnitTest.kt:20]:
```suggestion
```

File: base/src/test/java/com/mindera/skeletoid/threads/threadpools/ScheduledThreadPoolUnitTest.kt [hunk 6]
```diff
@@ -3,8 +3,12 @@ package com.mindera.skeletoid.threads.threadpools
 import com.mindera.skeletoid.utils.extensions.mock
 import org.junit.Assert
 import org.junit.Test
+import org.mockito.Mockito
+import java.util.concurrent.ArrayBlockingQueue
```
Comment (bruno-garces) [inline base/src/test/java/com/mindera/skeletoid/threads/threadpools/ScheduledThreadPoolUnitTest.kt]: unused?

File: base/src/test/java/com/mindera/skeletoid/webview/cookies/WebViewCookiesTest.kt [hunk 7]
```diff
@@ -41,20 +37,24 @@ class WebViewCookiesTest {
     @Test
     fun testClearWebViewCookiesApiEqualOrGreaterThan22() {
         Whitebox.setInternalState(Build.VERSION::class.java, "SDK_INT", 27)
+        //Move to new ApplicationProvider.getApplicationContext() throws an error
+        //probably need to use https://stackoverflow.com/questions/32957741/android-illegalstateexception-no-instrumentation-registered-must-run-under-a-re
```
Comment (bruno-garces) [inline base/src/test/java/com/mindera/skeletoid/webview/cookies/WebViewCookiesTest.kt]: Need to use what?

File: kt-extensions/src/main/java/com/mindera/skeletoid/kt/extensions/utils/Parcel.kt [hunk 8]
```diff
@@ -16,10 +16,6 @@ inline fun <reified T> parcelableCreator(
             override fun newArray(size: Int) = arrayOfNulls<T>(size)
         }
 
-fun Parcel.readBoolean() = readInt() != 0
-
-fun Parcel.writeBoolean(value: Boolean) = writeInt(if (value) 1 else 0)
```
Comment (bruno-garces) [inline kt-extensions/src/main/java/com/mindera/skeletoid/kt/extensions/utils/Parcel.kt:21]: not needed anymore?

File: base/src/main/java/com/mindera/skeletoid/threads/threadpools/ScheduledThreadPoolExecutor.java [hunk 9]
```diff
@@ -65,15 +65,13 @@ private void changeThreadsNameAfterShutdown() {
         final NamedThreadFactory factory = (NamedThreadFactory) getThreadFactory();
         if (factory != null) {
             final Queue<Thread> threads = factory.getThreads();
-            if (threads != null) {
-                for (Thread t : threads) {
-                    final String threadName = t.getName();
-                    if (threadName != null && !threadName.startsWith(SHUTDOWN_THREAD)) {
-                        t.setName(SHUTDOWN_THREAD + " " + t.getName());
-                    }
+            for (Thread t : threads) {
```
Comment (bruno-garces) [inline base/src/main/java/com/mindera/skeletoid/threads/threadpools/ScheduledThreadPoolExecutor.java]: meaningful name please

**Review by bruno-garces — CHANGES_REQUESTED** [review]

File: base/src/test/java/com/mindera/skeletoid/webview/cookies/WebViewCookiesTest.kt (hunk 7 above)
Comment (neteinstein) [reply base/src/test/java/com/mindera/skeletoid/webview/cookies/WebViewCookiesTest.kt]: `ApplicationProvider.getApplicationContext()` above

File: kt-extensions/src/main/java/com/mindera/skeletoid/kt/extensions/utils/Parcel.kt (hunk 8 above)
Comment (neteinstein) [reply kt-extensions/src/main/java/com/mindera/skeletoid/kt/extensions/utils/Parcel.kt:21]: Yep, the Parcel now has this exact implementation.

File: base/src/test/java/com/mindera/skeletoid/threads/threadpools/ScheduledThreadPoolUnitTest.kt (hunk 6 above)
Comment (bruno-garces) [reply base/src/test/java/com/mindera/skeletoid/threads/threadpools/ScheduledThreadPoolUnitTest.kt]: Did you forget this?

**Review by bruno-garces — APPROVED** [review]

File: base/src/main/java/com/mindera/skeletoid/threads/threadpools/NamedThreadFactory.kt (hunk 1 above)
Comment (bruno-garces) [reply base/src/main/java/com/mindera/skeletoid/threads/threadpools/NamedThreadFactory.kt]: ☝️

File: base/src/main/java/com/mindera/skeletoid/threads/threadpools/ScheduledThreadPoolExecutor.java (hunk 9 above)
Comment (bruno-garces) [reply base/src/main/java/com/mindera/skeletoid/threads/threadpools/ScheduledThreadPoolExecutor.java]: ☝️

File: base/src/test/java/com/mindera/skeletoid/webview/cookies/WebViewCookiesTest.kt (hunk 7 above)
Comment (bruno-garces) [reply base/src/test/java/com/mindera/skeletoid/webview/cookies/WebViewCookiesTest.kt]: Can we improve the comment?

```
// Moving to the new ApplicationProvider.getApplicationContext() throws an error
// We probably need to use (explain here what) like suggested in this SO thread
// https://stackoverflow.com/questions/32957741/android-illegalstateexception-no-instrumentation-registered-must-run-under-a-re
```

**Comment by neteinstein** [comment]
This is for some reason pending status, but it has passed on Travis: https://travis-ci.org/github/Mindera/skeletoid/builds/708709784

merging!

**Pull request merged**